### PR TITLE
[Correlation log] Add header(correlation id) to API response

### PR DESCRIPTION
**Pull request by sachithKay**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/rest/API.java [hunk 1]
```diff
@@ -313,6 +316,17 @@ void process(MessageContext synCtx) {
 			((Axis2MessageContext) synCtx).getAxis2MessageContext().
 							setProperty(NhttpConstants.REST_URL_POSTFIX,restURLPostfix);
 		}
+        if (synCtx.isResponse()){
+            org.apache.axis2.context.MessageContext context = ((Axis2MessageContext) synCtx).getAxis2MessageContext();
+            //SourceConfiguration sourceConfiguration = (SourceConfiguration)context.getProperty("PASS_THROUGH_SOURCE_CONFIGURATION");
```
Comment (isudana) [inline modules/core/src/main/java/org/apache/synapse/rest/API.java]: please remove commented code segment.

File: modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java [hunk 2]
```diff
@@ -84,7 +84,7 @@ public SynapseCallbackReceiver(SynapseConfiguration synCfg,
 
         // create the Timer object and a TimeoutHandler task
         TimeoutHandler timeoutHandler = new TimeoutHandler(callbackStore, contextInformation);
-        
+
```
Comment (isudana) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java]: Shall we remove unnecessary formatting changes from this PR ?

File: modules/core/src/main/java/org/apache/synapse/rest/API.java [hunk 3]
```diff
@@ -313,6 +316,16 @@ void process(MessageContext synCtx) {
 			((Axis2MessageContext) synCtx).getAxis2MessageContext().
 							setProperty(NhttpConstants.REST_URL_POSTFIX,restURLPostfix);
 		}
+        if (synCtx.isResponse()){
```
Comment (malakaganga) [inline modules/core/src/main/java/org/apache/synapse/rest/API.java]: Shall we fix the formatting issue here?

Comment (sachithKay) [reply modules/core/src/main/java/org/apache/synapse/rest/API.java]: fixed in https://github.com/wso2/wso2-synapse/pull/1305
